src/memusage.h

```diff
@@ -84,8 +84,8 @@ struct stl_shared_counter
     size_t weak_count;
 };
 
-template<typename X>
-static inline size_t DynamicUsage(const std::vector<X>& v)
+template<typename X, typename Y>
```

Comment:
right-it can be used for nested structures, but the call site has to do their own iteration and add it up, as this only covers the outermost layer

adding the `std::is_trivial_v` check is an interesting idea but would prevent it from being used there.

Reply:
In a next PR we could add another DynamicUsage which accepts a vector of vectors and does the iteration instead of the call sites